test-suite/src/test/scala/org/scalajs/testsuite/niobuffer/BufferAdapter.scala

```diff
@@ -0,0 +1,196 @@
+package org.scalajs.testsuite.niobuffer
+
+import java.nio._
+
+abstract class BufferAdapter[BT <: Buffer, ET] {
+  type BufferType = BT
+  type ElementType = ET
+
+  def slice(): BufferType
+  def duplicate(): BufferType
+  def asReadOnlyBuffer(): BufferType
+  def get(): ElementType
+  def put(e: ElementType): BufferType
+  def get(index: Int): ElementType
+  def put(index: Int, e: ElementType): BufferType
+  def get(dst: Array[ElementType], offset: Int, length: Int): BufferType
+  def get(dst: Array[ElementType]): BufferType
+  def put(src: BufferType): BufferType
+  def put(src: Array[ElementType], offset: Int, length: Int): BufferType
+  def put(src: Array[ElementType], dummy: Boolean = false): BufferType
```

Comment:
I think the canonical way to do this, is to have an implicit parameter list with a [DummyImplicit](http://www.scala-lang.org/api/current/index.html#scala.Predef$$DummyImplicit). Any reason you chose to do otherwise?